crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -6619,19 +6616,48 @@ impl<'db> TypeInferenceBuilder<'db> {
 
         builder = builder.add_positive(KnownClass::Bool.to_instance(self.db()));
 
+        let mut has_success = None;
+        let mut first_error = None;
+
         for pos in intersection.positive(self.db()) {
             let result = match intersection_on {
-                IntersectionOn::Left => {
-                    self.infer_binary_type_comparison(*pos, op, other, range)?
+                IntersectionOn::Left => self.infer_binary_type_comparison(*pos, op, other, range),
+                IntersectionOn::Right => self.infer_binary_type_comparison(other, op, *pos, range),
+            };
+
+            match result {
+                Ok(ty) => {
+                    has_success = Some(true);
+                    builder = builder.add_positive(ty);
                 }
-                IntersectionOn::Right => {
-                    self.infer_binary_type_comparison(other, op, *pos, range)?
+                Err(error) => {
+                    has_success = has_success.or(Some(false));
+                    first_error = first_error.or(Some(error));
                 }
-            };
-            builder = builder.add_positive(result);
+            }
         }
 
-        Ok(builder.build())
+        match has_success {
+            Some(true) | None => Ok(builder.build()),
```

Comment:
I think we should actually show an error if we saw no element that supports the operation.